sdk/trace/src/test/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessorTest.java

```diff
@@ -378,48 +371,57 @@ void exportNotSampledSpans() {
     // have a span1 variable.
-    assertThat(exported).containsExactly(span2.toSpanData());
+    assertThat(exported).containsExactly(span.toSpanData());
   }
 
   @Test
-  void exportNotSampledSpans_recordingEvents() {
-    // TODO(bdrutu): Fix this when Sampler return RECORD_ONLY option.
-    /*
-    sdkTracerProvider.addSpanProcessor(
-        BatchSpanProcessor.builder(waitingSpanExporter)
-            .setScheduleDelayMillis(MAX_SCHEDULE_DELAY_MILLIS)
-            .reportOnlySampled(false)
-            .build());
+  void exportNotSampledSpans_recordOnly() {
+    WaitingSpanExporter waitingSpanExporter =
+        new WaitingSpanExporter(1, CompletableResultCode.ofSuccess());
+    AtomicReference<TraceConfig> traceConfig =
+        new AtomicReference<>(
+            TraceConfig.getDefault().toBuilder()
+                .setSampler(
+                    new Sampler() {
+                      @Override
+                      public SamplingResult shouldSample(
+                          Context parentContext,
+                          String traceId,
+                          String name,
+                          Span.Kind spanKind,
+                          Attributes attributes,
+                          List<SpanData.Link> parentLinks) {
+                        return SamplingResult.create(SamplingResult.Decision.RECORD_ONLY);
+                      }
 
-    ReadableSpan span = createNotSampledRecordingEventsEndedSpan(SPAN_NAME_1);
-    List<SpanData> exported = waitingSpanExporter.waitForExport(1);
-    assertThat(exported).containsExactly(span.toSpanData());
-    */
-  }
+                      @Override
+                      public String getDescription() {
+                        return null;
+                      }
+                    })
+                .build());
+    sdkTracerProvider =
+        SdkTracerProvider.builder()
+            .addSpanProcessor(
+                BatchSpanProcessor.builder(waitingSpanExporter)
+                    .setScheduleDelayMillis(MAX_SCHEDULE_DELAY_MILLIS)
+                    .setExportOnlySampled(true)
+                    .build())
+            .setTraceConfig(traceConfig::get)
+            .build();
 
-  @Test
-  void exportNotSampledSpans_reportOnlySampled() {
-    // TODO(bdrutu): Fix this when Sampler return RECORD_ONLY option.
-    /*
-    sdkTracerProvider.addSpanProcessor(
-        BatchSpanProcessor.builder(waitingSpanExporter)
-            .reportOnlySampled(true)
-            .setScheduleDelayMillis(MAX_SCHEDULE_DELAY_MILLIS)
-            .build());
-
-    createNotSampledRecordingEventsEndedSpan(SPAN_NAME_1);
-    ReadableSpan sampledSpan = createSampledEndedSpan(SPAN_NAME_2);
-    List<SpanData> exported = waitingSpanExporter.waitForExport(1);
-    assertThat(exported).containsExactly(sampledSpan.toSpanData());
-    */
+    createEndedSpan(SPAN_NAME_2);
+
+    List<SpanData> exported = waitingSpanExporter.waitForExport();
```

Comment:
I think you need to do a similar pattern as the other test of exporting a sampled span after the record only span and checking it's the only exported span. Otherwise, this isn't doing any waiting I believe since nothing to wait for.

Reply:
The exporter is waiting for 1, and I was relying on the timeout to ensure nothing was getting through. Am I misunderstanding how the waiting exporter works?